### PR TITLE
Fix card component stale state on dropin component reuse

### PR DESCRIPTION
**Pull request by atmamont**
# Summary

Every time certain components are reused within same Dropin session, it has stale parts like:
- error messages for gift card component
- brand icons for card component

This fix adds:
- missing `reset()` call for `FormErrorItemView` (https://github.com/Adyen/adyen-ios/pull/1920/commits/3bf58e0de5d8c3520297607df90f4f0de9c64c7c)
- missing nullification for detected brands logos urls for `FormCardNumberItemView` (https://github.com/Adyen/adyen-ios/pull/1920/commits/d24f7d7df82dfbb6d249bbd71a51ea012e11be61)

It is recommended to review by commits provided.

# Release notes

<fixed> 
For Drop-in, some components that are used in the same Drop-in session no longer have stale parts such as the following:   
- For gift cards, error messages.  
- For cards, brand logos.  
</fixed>

# Ticket

<ticket>
COIOS-840
</ticket>

### CODE REVIEW

File: Adyen/UI/Form/Items/Error/FormErrorItemView.swift [hunk 1]
```diff
@@ -80,4 +80,10 @@ internal final class FormErrorItemView: FormItemView<FormErrorItem> {
         return view
     }()
 
+    override internal func reset() {
```
Comment (erenbesel) [inline Adyen/UI/Form/Items/Error/FormErrorItemView.swift:83]: NIce find!

File: AdyenCard/Form/DualBrandView.swift [hunk 2]
```diff
@@ -90,6 +90,9 @@ extension FormCardNumberItemView {
             guard let firstLogo = logos.first else { return }
             let secondLogo = logos.adyen[safeIndex: 1]
             
+            primaryLogoUrl = firstLogo.url
```
Comment (erenbesel) [inline AdyenCard/Form/DualBrandView.swift:93]: Tried to look commit by commit but I see these 2 lines were already in the code, why is it shown as newly added?

Comment (atmamont) [reply AdyenCard/Form/DualBrandView.swift:93]: This is my bad, I think I commited these two lines when was extracting this to a separate file 🙈

Comment (atmamont) [reply AdyenCard/Form/DualBrandView.swift:93]: It was like this before 
image

Comment (atmamont) [reply AdyenCard/Form/DualBrandView.swift:93]: Good catch, I missed the right method 🙈 
Updated and force pushed, check [this commit ](https://github.com/Adyen/adyen-ios/pull/1920/commits/d24f7d7df82dfbb6d249bbd71a51ea012e11be61)

**Review by erenbesel — APPROVED** [review]

**Review by nauaros — DISMISSED** [review]

**Review by erenbesel — DISMISSED** [review]
Looks good! Maybe we can add some tests? seems we don't have any

**Comment by atmamont** [comment]
> Looks good! Maybe we can add some tests? seems we don't have any

👌 
In progress, DualBandView has zero coverage now so I'll add more apart from those that check this specific fix.

**Comment by atmamont** [comment]
@goergisn @erenbesel Unit tests added, can I have another look?

File: Tests/IntegrationTests/Card Tests/DualBrandViewTests.swift [hunk 3]
```diff
@@ -0,0 +1,162 @@
...
+    }
+    
+    override func tearDown() {
+        sut = nil
+        imageLoader = nil
+        brandSelectionCount = nil
+        super.tearDown()
+    }
+    
+    func testUpdateCurrentLogos_WhenResettingLoadedImages_ShouldResetToPlaceholder() {
```
Comment (atmamont) [inline Tests/IntegrationTests/Card Tests/DualBrandViewTests.swift:36]: Let me know if this guy looks too big and it is not easy to follow. If so, I'll break it to 3-4 separate tests.

File: Tests/IntegrationTests/Card Tests/DualBrandViewTests.swift [hunk 4]
```diff
@@ -0,0 +1,162 @@
...
+        borderColor: nil,
+        borderWidth: 0.0,
+        cornerRadius: 0.0,
+        clipsToBounds: true,
+        contentMode: .scaleAspectFit
+    )
+
+}
+
+private class MockImageLoader: ImageLoading {
```
Comment (erenbesel) [inline Tests/IntegrationTests/Card Tests/DualBrandViewTests.swift]: nice!

**Review by erenbesel — DISMISSED** [review]

**Review by goergisn — DISMISSED** [review]

File: Tests/IntegrationTests/Card Tests/DualBrandViewTests.swift (hunk 4 above)
Comment (goergisn) [reply Tests/IntegrationTests/Card Tests/DualBrandViewTests.swift]: we already have an `ImageLoaderMock` - maybe that can be adapted instead if needed?

Comment (atmamont) [reply Tests/IntegrationTests/Card Tests/DualBrandViewTests.swift]: Good catch, indeed there is no need to create duplicates

**Review by erenbesel — APPROVED** [review]

**Review by goergisn — APPROVED** [review]

**Pull request merged**